tests/extensions_class_test.py

```diff
@@ -0,0 +1,61 @@
+import os
+import shutil
+import tempfile
+import unittest
```

Comment:
Is there a reason we are switching away from pytest to use unitest instead?

Reply:
just a preference, they both work well together anyway. 

I like the `setUp` / `tearDown` of unittest

Reply:
Sure, but we then have another dependancy and another module requirement.

Reply:
it comes packaged with Python afaik

Reply:
Yes, `unittest` is packaged with Python. `pytest` is, essentially, just a 3rd party unittest framework, with the intention to make writing tests easier.

Reply:
Ok, so we are switching to unittest away from pytest? I just don't want to make a mess and have multiple test suites. The reason I redid my original unittest tests and switched to pytest was that its much easier to understand, especially when working with things like emulating a web api and for using fixtures.

Reply:
Oh it's the opposite for me, I find `pytest` fixtures very confusing. 

I've used `unittest` in `devkit`(https://github.com/area28technologies/devkit/blob/master/tests/build_test.py) and I find stuff like `f'{os.path.dirname(__file__)}/fixtures/fake-pkg'` more clear than some decorators and some hidden magic with folder names.

But we could do it in `pytest` too.

And for emulating API it's not related to the test suite. I've used https://github.com/getsentry/responses in devkit seems easier and more used than https://github.com/csernazs/pytest-httpserver 

I actually don't really mind either, I'll just rewrite them in `pytest`  👍